posthog/hogql/functions/mapping.py

```diff
@@ -965,6 +965,283 @@ def compare_types(arg_types: list[ConstantType], sig_arg_types: tuple[ConstantTy
+        "formatDateTime",
+        2,
+        3,
+        tz_aware=True,
+        signatures=[
+            ((DateTimeType(), StringType()), StringType()),
+            ((DateTimeType(), StringType(), StringType()), StringType()),
+        ],
+    ),
+    "make_timestamp": HogQLFunctionMeta(
+        "makeDateTime",
+        6,
+        7,
+        tz_aware=True,
+        signatures=[
+            ((IntegerType(), IntegerType(), IntegerType(), IntegerType(), IntegerType(), FloatType()), DateTimeType()),
+            (
+                (IntegerType(), IntegerType(), IntegerType(), IntegerType(), IntegerType(), FloatType(), StringType()),
+                DateTimeType(),
+            ),
+        ],
+    ),
+    "make_date": HogQLFunctionMeta(
+        "makeDate",
+        3,
+        3,
+        signatures=[
+            ((IntegerType(), IntegerType(), IntegerType()), DateType()),
+        ],
+    ),
+    "timezone": HogQLFunctionMeta(
+        "toTimeZone",
+        2,
+        2,
+        tz_aware=True,
+        signatures=[
+            ((StringType(), DateTimeType()), DateTimeType()),
+        ],
+    ),
+    "now": HogQLFunctionMeta(
+        "now64",
+        0,
+        1,
+        tz_aware=True,
+        case_sensitive=False,
+        signatures=[
+            ((), DateTimeType()),
+            ((UnknownType(),), DateTimeType()),
+        ],
+    ),
+    "current_timestamp": HogQLFunctionMeta(
+        "now64",
+        0,
+        0,
+        tz_aware=True,
+        signatures=[
+            ((), DateTimeType()),
+        ],
+    ),
+    "current_date": HogQLFunctionMeta(
+        "today",
+        0,
+        0,
+        signatures=[
+            ((), DateType()),
+        ],
+    ),
+    #  This doesn't work yet but will in a new version of Clickhouse: https://github.com/ClickHouse/ClickHouse/pull/56738
+    # "date_bin": HogQLFunctionMeta(
+    #     "toSTartOfInterval({1}, {0}, {2})",
+    #     3,
+    #     3,
+    #     tz_aware=True,
+    #     signatures=[
+    #         ((IntervalType(), DateTimeType(), DateTimeType()), DateTimeType()),
+    #     ],
+    # ),
+    "date_add": HogQLFunctionMeta(
+        "date_add",
+        2,
+        2,
+        tz_aware=True,
+        signatures=[
+            ((DateTimeType(), IntervalType()), DateTimeType()),
+        ],
+    ),
+    "date_subtract": HogQLFunctionMeta(
+        "date_sub",
+        2,
+        2,
+        tz_aware=True,
+        signatures=[
+            ((DateTimeType(), IntervalType()), DateTimeType()),
+        ],
+    ),
+    "date_diff": HogQLFunctionMeta(
+        "dateDiff",
+        3,
+        3,
+        signatures=[
+            ((StringType(), DateTimeType(), DateTimeType()), IntegerType()),
+        ],
+    ),
+    "to_date": HogQLFunctionMeta(
+        "toDate",
+        1,
+        2,
+        signatures=[
+            ((StringType(),), DateType()),
+            ((StringType(), StringType()), DateType()),
+        ],
+    ),
```

Comment:
Same with all of these like above - maintaining two versions of the same function is only gonna get messy